modules/spring-data-commons/src/main/java/org/apache/ignite/internal/transactions/proxy/ClientTransactionProxy.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.transactions.proxy;
+
+import org.apache.ignite.client.ClientTransaction;
+import org.apache.ignite.internal.util.typedef.internal.S;
+
+/**
+ * Represents {@link TransactionProxy} implementation that uses {@link ClientTransaction} to perform transaction
+ * operations.
+ */
+public class ClientTransactionProxy implements TransactionProxy {
```

Comment:
As far as I understand transaction proxies don't use anyhow cache data proxies and cache data proxies don't use anyhow transactions proxies. Transactions proxies required only by one module spring-tx, this module can be used without spring data, so I think we should move back these classes to spring-tx module.
Moreover, there is no dependency from spring-tx to spring-data-commons module.

Reply:
My motivation to put the tx proxy classes in spring-data-commons is to make it possible to reuse all the proxy classes.

In the future, I would like to rename spring-data-commons to spring-commons.

I think it is useful to keep all of the Spring utility classes in one module - for example, the IgniteCacheProxy, which is now stored in spring-data-commons, will be reused in  the Spring Cache thin client integration. But earlier I claimed that IgniteCacheProxy is only needed for Spring Data. The same can be with the tx proxy.

Does this make sense?

Reply:
Spring data proxies are used by 3 modules, so it makes sense to move them into the common module to reduce duplication. Spring-tx proxies only used by one. If in the future there will be new modules that will require spring-tx proxies, we can move these classes to the common module, but now I don't know modules except spring-tx that require spring-tx proxies. Did I miss something?

Reply:
I moved proxy classes back to tx-ext module.